src/sentry/api/helpers/group_index/update.py

```diff
@@ -619,6 +529,122 @@ def process_group_resolution(
+        release = status_details.get("inUpcomingRelease") or most_recent_release(projects[0])
+        activity_type = ActivityType.SET_RESOLVED_IN_RELEASE.value
+        activity_data = {"version": ""}
+
+        serialized_user = user_service.serialize_many(filter=dict(user_ids=[user.id]), as_user=user)
+        new_status_details = {
+            "inUpcomingRelease": True,
+        }
+        if serialized_user:
+            new_status_details["actor"] = serialized_user[0]
+        res_type = GroupResolution.Type.in_upcoming_release
+        res_type_str = "in_upcoming_release"
+        res_status = GroupResolution.Status.pending
+    elif status_details.get("inRelease"):
+        release = status_details["inRelease"]
+        activity_type = ActivityType.SET_RESOLVED_IN_RELEASE.value
+        activity_data = {
+            # no version yet
+            "version": release.version
+        }
+
+        serialized_user = user_service.serialize_many(filter=dict(user_ids=[user.id]), as_user=user)
+        new_status_details = {
+            "inRelease": release.version,
+        }
+        if serialized_user:
+            new_status_details["actor"] = serialized_user[0]
+        res_type = GroupResolution.Type.in_release
+        res_type_str = "in_release"
+        res_status = GroupResolution.Status.resolved
+    elif status_details.get("inCommit"):
+        commit = status_details["inCommit"]
+        activity_type = ActivityType.SET_RESOLVED_IN_COMMIT.value
+        activity_data = {"commit": commit.id}
+        serialized_user = user_service.serialize_many(filter=dict(user_ids=[user.id]), as_user=user)
+
+        new_status_details = {
+            "inCommit": serialize(commit, user),
+        }
+        if serialized_user:
+            new_status_details["actor"] = serialized_user[0]
+        res_type_str = "in_commit"
+    else:
+        res_type_str = "now"
+        activity_type = ActivityType.SET_RESOLVED.value
+        activity_data = {}
+        new_status_details = {}
+
+    # if we've specified a commit, let's see if its already been released
+    # this will allow us to associate the resolution to a release as if we
+    # were simply using 'inRelease' above
+    # Note: this is different than the way commit resolution works on deploy
+    # creation, as a given deploy is connected to an explicit release, and
+    # in this case we're simply choosing the most recent release which contains
+    # the commit.
+    if commit and not release:
+        # TODO(jess): If we support multiple projects for release / commit resolution,
+        # we need to update this to find the release for each project (we shouldn't assume
+        # it's the same)
+        try:
+            release = most_recent_release_matching_commit(projects, commit)
+            res_type = GroupResolution.Type.in_release
+            res_status = GroupResolution.Status.resolved
+        except IndexError:
+            release = None
```

Comment:
These lines come from this block:
https://github.com/getsentry/sentry/blob/7efb91bbb24f72bc0b1db7356ad89bef4d8016a3/src/sentry/api/helpers/group_index/update.py#L407-L423